sig-architecture/production-readiness.md

```diff
@@ -41,6 +41,62 @@ aspects of the process.
   with and without the feature are necessary. At the very least, think about
   conversion tests if API types are being modified.
 
+
+#### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g. what if some components will restart
+  in the middle of rollout?
+
+* **What signals should trigger rollback?**
+  Focus on metrics and treat other signals (like logs) as last resort.
+
+* **Were upgrade and rollback tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and do that now.
+
+
+#### Monitoring requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
```

Comment:
I think we can require this to be a metric, not just logging or events.

Reply:
Partially agree. Logging or events are bad. But if metrics are hard for some reason, I can imagine situations like:
- the use of a feature is reflected in a specific value of some field (e.g. in a Pod)
- kubectl + some output parsing may potentially be an option to determine it 

Agree that metrics are preferred though. Reflected in the doc - let me know if you're fine with it.